cpp/generate_issn_to_titles_and_ppns_map.cc

```diff
@@ -42,45 +44,79 @@ std::string EscapeColons(const std::string &unescaped_string) {
-    unsigned processed_count(0), generated_count(0);
+auto SplitAndDedupeEntries(const std::string &entries) {
+    std::vector<std::string> individual_entries;
+    StringUtil::Split(entries, ' ', &individual_entries);
+    std::set<std::string> deduplicated_entries;
+    deduplicated_entries.insert(individual_entries.cbegin(), individual_entries.cend());
+    return deduplicated_entries;
+}
+
 
-    while (const auto record = reader->read()) {
-        ++processed_count;
+unsigned ProcessZederAndWriteMapFile(File * const output, const Zeder::SimpleZeder &zeder) {
+    unsigned generated_count(0);
+    for (const auto &journal : zeder) {
+        if (journal.empty())
+            continue;
 
-        if (not record.isSerial())
+        const auto print_issns(SplitAndDedupeEntries(journal.lookup("issn")));
+        if (print_issns.empty())
             continue;
 
-        const auto issns(record.getISSNs());
-        if (issns.empty())
+        const auto electronic_issns(SplitAndDedupeEntries(journal.lookup("essn")));
+        if (electronic_issns.empty())
             continue;
 
-        const auto title(record.getMainTitle());
+        const auto title(journal.lookup("tit"));
         if (title.empty())
             continue;
 
-        const auto ppn(record.getControlNumber());
-        for (const auto issn : issns) {
-            (*output) << issn << ':' << EscapeColons(title) << ':' << ppn << '\n';
-            ++generated_count;
+        const auto electronic_ppns(SplitAndDedupeEntries(journal.lookup("eppns")));
+        if (electronic_ppns.empty() or electronic_ppns.size() != electronic_issns.size())
+            continue;
+
+        for (const auto print_issn : print_issns) {
+            auto electronic_issn(electronic_issns.cbegin());
+            auto electronic_ppn(electronic_ppns.cbegin());
+            while (electronic_issn != electronic_issns.cend()) {
+                (*output) << print_issn << ':' << EscapeColons(title) << ':' << (*electronic_issn) << ':' << (*electronic_ppn) << '\n';
+                ++generated_count;
+
+                ++electronic_issn, ++electronic_ppn;
+            }
         }
     }
 
-    LOG_INFO("Processed " + std::to_string(processed_count) + " MARC records and generated " + std::to_string(generated_count) + " map entry/entries.");
+    return generated_count;
+}
+
+
+unsigned ProcessZederFlavour(const Zeder::Flavour zeder_flavour, File * const map_output) {
+    const Zeder::SimpleZeder zeder(zeder_flavour, { "eppns", "essn", "issn", "tit" });
+    if (unlikely(zeder.empty()))
+        LOG_ERROR("found no IxTheo Zeder entries matching any of our requested columns!");
+    return ProcessZederAndWriteMapFile(map_output, zeder);
 }
 
 
 } // unnamed namespace
 
 
 int Main(int argc, char **argv) {
-    if (argc != 3)
-        ::Usage("marc_input mapfile_output");
+    if (argc != 2)
+        ::Usage("mapfile_output");
+
+
+
+    const auto temp_file(FileUtil::OpenTempFileOrDie("/tmp/XXXXXX"));
```

Comment:
Replacing this by `FileUtil::AutoTempFile temp_file;` might be shorter

Reply:
No it isn't because AutoTempFile only gives us a std::string and we need a File *.